### PR TITLE
Update AccessController to return appropriate errors when the OpenID settings are invalid

### PR DESCRIPTION
**Pull request by PinpointTownes**
Fixes https://github.com/OrchardCMS/Orchard2/issues/464.

### CODE REVIEW

File: src/Orchard.Cms.Web/Modules/Orchard.OpenId/Controllers/AccessController.cs [hunk 1]
```diff
@@ -267,15 +294,25 @@ public class AccessController : Controller
                 });
             }
 
+            var request = HttpContext.GetOpenIdConnectRequest();
```
Comment (sebastienros) [inline src/Orchard.Cms.Web/Modules/Orchard.OpenId/Controllers/AccessController.cs]: Could use two methods like:

```
If (!CheckConfiguration()) 
{
  return ConfigurationError();
}
```
If not just for the error message duplication

Comment (PinpointTownes) [inline src/Orchard.Cms.Web/Modules/Orchard.OpenId/Controllers/AccessController.cs]: The `CheckConfiguration()` thingy won't work, as we need the `request` instance in the action (it was previously provided as an action parameter).

Concerning `ConfigurationError`, it seems a bit ugly, but if you believe it's better, I'm fine changing it.

Comment (sebastienros) [inline src/Orchard.Cms.Web/Modules/Orchard.OpenId/Controllers/AccessController.cs]: > as we need the request instance 

You mean, a parameter?

> it seems a bit ugly

Whatever you think is less ugly but would reuse the BadRequest code and message.

Comment (PinpointTownes) [inline src/Orchard.Cms.Web/Modules/Orchard.OpenId/Controllers/AccessController.cs]: > You mean, a parameter?

It's not really a parameter, at least not _stricto sensu_: `OpenIdConnectRequest` is populated by ASOS as part of the request extraction process, using either the query string or the request body (depending on the type of the request and whether GET or POST was used). It's then stored in a special feature named `OpenIdConnectServerFeature` that holds both the OIDC request and the response objects. `GetOpenIdConnectRequest()` is a wrapper around that stuff.

> Whatever you think is less ugly but would reuse the BadRequest code and message.

`BadRequest` is only used here ('cause it's an "API action"). The other actions return a view when something is wrong.

Comment (sebastienros) [inline src/Orchard.Cms.Web/Modules/Orchard.OpenId/Controllers/AccessController.cs]: Good then, thanks

**Comment by PinpointTownes** [comment]
Thanks for the review :smile:

**Pull request merged**